server/community_test.go

```diff
@@ -0,0 +1,69 @@
+package server
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/google/go-github/v32/github"
+	"github.com/mattermost/mattermost-mattermod/model"
+	"github.com/mattermost/mattermost-mattermod/server/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPostPRWelcomeMessage(t *testing.T) {
+	pr := &model.PullRequest{
+		RepoOwner: "owner",
+		RepoName:  "repoName",
+		Number:    123,
+		Username:  "foo",
+	}
+
+	ctxInterface := reflect.TypeOf((*context.Context)(nil)).Elem()
+
+	for name, test := range map[string]struct {
+		SetupClient func(*gomock.Controller) *GithubClient
+		OrgMembers  []string
+	}{
+		"No org member": {
+			SetupClient: func(ctrl *gomock.Controller) *GithubClient {
+				issueMocks := mocks.NewMockIssuesService(ctrl)
+				client := &GithubClient{
+					Issues: issueMocks,
+				}
+
+				comment := &github.IssueComment{Body: github.String("Hi @foo, thanks for the PR!")}
+
+				issueMocks.EXPECT().CreateComment(gomock.AssignableToTypeOf(ctxInterface),
+					gomock.Eq(pr.RepoOwner), gomock.Eq(pr.RepoName),
+					gomock.Eq(pr.Number), comment).Return(nil, nil, nil)
+
+				return client
+			},
+			OrgMembers: []string{"bar"},
+		},
+		"Org member": {
+			SetupClient: func(ctrl *gomock.Controller) *GithubClient {
+				return &GithubClient{}
+			},
```

Comment:
I wonder if it would makes sense to use mocks with `0` expected calls.
```suggestion
			SetupClient: func(ctrl *gomock.Controller) *GithubClient {
				issueMocks := mocks.NewMockIssuesService(ctrl)
				client := &GithubClient{
					Issues: issueMocks,
				}

				return client
			},
```

Reply:
What is the advantage? Avoid nil pointers in tests?

Reply:
Yes, and also makes sure that we don't receive unexpected calls. If receive any e.g. `(*GitHubClient).SendGitHubComment`, we can take a look since this test will going to fail.